src/python/CMSMonitoring/StompAMQ.py

```diff
@@ -44,17 +47,19 @@ def validate(doc, schema='auto', verbose=False):
                 logging.warning('Local schema {} is not json compliant'.format(schema))
 
     if schema in _local_schemas:
-        return validate_schema(_local_schemas[schema], doc, verbose)
+        return validate_schema(_local_schemas[schema], doc, loglevel)
 
     elif schema in _schemas.schemas():
-        return validate_schema(_schemas.schemas()[schema], doc, verbose)
+        return validate_schema(_schemas.schemas()[schema], doc, loglevel)
 
     else:
-        for sname, sch in _schemas.schemas().items():
-            if validate_schema(sch, doc, verbose):
-                return True
+        logging.warning('No validation schema provided, comparing all docs to the first one')
```

Comment:
This part will not work in general. Here is an example using WMAgent:
- client didn't provide a schema
- it forces to create a file `auto_schema.json`
- then client validate doc1 which confirm with this schema
- but then client use another doc which does not belong to the schema, e.g. WMA may have different type of docs, according to Alan they may have around 30 different schemas. At this point the validation will fail since it relies on doc1 schema which may not be applicable to other docs.

I also dislike creation of `auto_schema.json` files at random place (this is what code implies) since such documents will be created at run time environment there is no procedure to clean-up them.

There is also no connection how to put this schema back into repository and therefore we will not have track of what has been validated. Having schemas in one repository which we can tag provides us with a way to trace down which schemas (versions) was used in particular period of time.

My suggestion to get rid of auto complete and force users to specify the schema name. All schemas will be part of repository and API should either provide existing file name (which we can check with `os.path.exists`) or file name which should match existing schemas from repository.

Reply:
As I wrote above, this is meant as a somewhat more sensible default behavior, but obviously users should be providing a schema and not rely on this. I'm happy to remove it altogether.

On the other hand, it doesn't really make a difference. Someone using this will quickly run into validation failures and can then create a schema and use that.

If you insist, I'm happy to remove it altogether.